src/test/kotlin/org/rust/lang/core/type/RsDivergingTypeInferenceTest.kt

```diff
@@ -1,45 +0,0 @@
-/*
- * Use of this source code is governed by the MIT license that can be
- * found in the LICENSE file.
- */
-
-package org.rust.lang.core.type
-
-
-class RsDivergingTypeInferenceTest : RsTypificationTestBase() {
-    fun `test if with all branches diverging`() = testExpr("""
-        fn main() {
-            let x = if true { return; } else { return; };
```

Comment:
This now infers to `unknown` instead of `!`.
Is this intentional?

Also you seem to have changed the test in a way that this change in behavior is not visible...
Is it intentional that the tests now test something different (the whole block instead of only `x`)?

Reply:
Yes. I think rustc can't infer the type of `if true { return; } else { return; }`

Reply:
I am pretty sure rust can infer the type as `!` as the following code compiles
```rust
fn main() {
    let x = if true { return; } else { return; };
    let y: String = x; //only possible because x is !
}
```

Reply:
Well, I still don't think rustc could infer a proper type for `if true { return; } else { return; }` by itself after some tests.

I tried run `rustc test.rs -Z save-analysis` and read the dumped json then, which gave me that the type of `x` is `std::string::String`, which's what the `y`'s type ascription says. And after I added `feature(never_type)` and set the type ascription of `x` to `!`, the type turned into `!` in the dumped json file.

And I got an empty funtion when I tried to dump mir of those code.

Reply:
I start to think `x` should be infer to `!` after reading rfc 1216 and its related comments. I'll fix this

Reply:
fixed